proposals/0345-android-12-issues.md

```diff
@@ -165,19 +268,67 @@ private static SdlDeviceListener getSdlDeviceListener(Context context, Bluetooth
 }
 ~~~
 
-We also need to consider interactions with applications supporting older versions of the `SdlRouterService`. If an application using SDL Android 4.11 or older receives an `ACL_CONNECTED` broadcast in the `SdlBroadcastReceiver`, it may try to start up the `SdlRouterService` of an application that is targeting Android 12. If the started `SdlRouterService` does not have Bluetooth permissions, it will not be able to use the `bluetoothTransport`. To solve this issue we can update the `initCheck` within the `SdlRouterService` to check the app's Bluetooth permissions. If the permissions are not granted, we can fail the `initCheck` and try to deploy the next `SdlRouterService`.
+We also need to consider interactions with applications supporting older versions of the `SdlRouterService`. If an application using SDL Android 4.11 or older receives an `ACL_CONNECTED` broadcast in the `SdlBroadcastReceiver`, it may try to start up the `SdlRouterService` of an application that is targeting Android 12. If the started `SdlRouterService` does not have Bluetooth permission, it will not be able to use the `bluetoothTransport`. To solve this issue we can update the `initCheck` within the `SdlRouterService` to check the app's Bluetooth permission. If the permission is not granted, we can fail the `initCheck` and try to deploy the next `SdlRouterService`.
 
-If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have Bluetooth permissions, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth permissions. By clicking on the notification the user will be directed to the app's settings page to grant those permissions. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted, the `bluetoothTransport` will be started.
+If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have Bluetooth permission, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth permission. By clicking on the notification the user will be directed to the app's settings page to grant this permission. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted, the `bluetoothTransport` will be started.
```

Comment:
```suggestion
If a phone connects to the head unit over USB we can still start the `SdlRouterService`, but if the designated app does not have the Bluetooth permission, the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable the Bluetooth permission. By clicking on the notification the user will be directed to the app's settings page to grant this permission. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permission is granted, the `bluetoothTransport` will be started.
```